deepchem/models/tensorgraph/layers.py

```diff
@@ -609,3 +677,262 @@ def _create_tensor(self):
     entropy, weights = self.in_layers[0], self.in_layers[1]
     self.out_tensor = tf.reduce_sum(entropy.out_tensor * weights.out_tensor)
     return self.out_tensor
+
+
+class AtomicConvolution(Layer):
+
+  def __init__(self,
+               atom_types=None,
+               radial_params=list(),
+               boxsize=None,
+               **kwargs):
+    """Atomic convoluation layer
+
+    N = max_num_atoms, M = max_num_neighbors, B = batch_size, d = num_features
+    l = num_radial_filters * num_atom_types
+
+    Parameters
+    ----------
+    
+    atom_types: list or None
+      Of length a, where a is number of atom types for filtering.
+    radial_params: list
+      Of length l, where l is number of radial filters learned.
+    boxsize: float or None
+      Simulation box length [Angstrom].
+    
+    """
+    self.boxsize = boxsize
+    self.radial_params = radial_params
+    self.atom_types = atom_types
+    super(AtomicConvolution, self).__init__(**kwargs)
+
+  def _create_tensor(self):
+    """
+    Parameters
+    ----------
+    X: tf.Tensor of shape (B, N, d)
+      Coordinates/features.
+    Nbrs: tf.Tensor of shape (B, N, M)
+      Neighbor list.
+    Nbrs_Z: tf.Tensor of shape (B, N, M)
+      Atomic numbers of neighbor atoms.
+      
+    
+    
+    Returns
+    -------
+    layer: tf.Tensor of shape (l, B, N)
+      A new tensor representing the output of the atomic conv layer 
+    """
+
+    X = self.in_layers[0].out_tensor
+    Nbrs = tf.to_int32(self.in_layers[1].out_tensor)
+    Nbrs_Z = self.in_layers[2].out_tensor
+
+    # N: Maximum number of atoms
+    # M: Maximum number of neighbors
+    # d: Number of coordinates/features/filters
+    # B: Batch Size
+    N = X.get_shape()[-2].value
+    d = X.get_shape()[-1].value
+    M = Nbrs.get_shape()[-1].value
+    B = X.get_shape()[0].value
+
+    D = self.distanceTensor(X, Nbrs, self.boxsize, B, N, M, d)
+    R = self.distanceMatrix(D)
+    sym = []
+    rsf_zeros = tf.zeros((B, N, M))
+    for param in self.radial_params:
+
+      # We apply the radial pooling filter before atom type conv
+      # to reduce computation
+      rsf = self.radialSymmetryFunction(R, *param)
+
+      if not self.atom_types:
+        cond = tf.not_equal(Nbrs_Z, 0.0)
+        sym.append(tf.reduce_sum(tf.where(cond, rsf, rsf_zeros), 2))
+      else:
+        for j in range(len(self.atom_types)):
+          cond = tf.equal(Nbrs_Z, self.atom_types[j])
+          sym.append(tf.reduce_sum(tf.where(cond, rsf, rsf_zeros), 2))
+
+    # Pack l (B, N) tensors into one (l, B, N) tensor
+    # Transpose to (B, N, l) for conv layer stacking
+    # done inside conv_layer loops to reduce transpose ops
+    # Final layer should be shape (N, B, l) to pass into tf.map_fn
+    # TODO (LESWING) batch norm
+    self.out_tensor = tf.stack(sym)
+    return self.out_tensor
+
+  def radialSymmetryFunction(self, R, rc, rs, e):
```

Comment:
Could you change this to `def radial_symmetry_function(...)`? We haven't used camel-case so far in DeepChem.